src/main/java/org/ethereum/beacon/discovery/packet/TaggedMessage.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.ethereum.beacon.discovery.packet;
+
+import java.util.List;
+import org.apache.tuweni.bytes.Bytes;
+import org.web3j.rlp.RlpDecoder;
+import org.web3j.rlp.RlpEncoder;
+import org.web3j.rlp.RlpString;
+import org.web3j.rlp.RlpType;
+
+/**
+ * Any message prepended with 'tag' and 'authTag' RLP
+ *
+ * <p>Format:<code>
+ * tagged-message = tag || rlp_bytes(auth-tag) || message-payload
+ * auth-tag = 12 random bytes unique to message
+ * message-payload = any data</code>
+ */
+class TaggedMessage {
+  public static final int TAG_SIZE = 32;
+  public static final int RLP_AUTH_TAG_SIZE = 12 + 1;
+  public static final int HEADER_SIZE = TAG_SIZE + RLP_AUTH_TAG_SIZE;
+
+  private final Bytes tag;
+  private final Bytes authTag;
+  private final Bytes payload;
+
+  public static TaggedMessage decode(Bytes bytes) {
+    Bytes tag = Bytes.wrap(bytes.slice(0, TAG_SIZE));
+    Bytes authTagRlpBytes = bytes.slice(TAG_SIZE, RLP_AUTH_TAG_SIZE);
+    Bytes authTag = decodeSingleRlpString(authTagRlpBytes);
+    if (authTag.size() != 12) {
```

Comment:
Since you went to the trouble of defining constants, why not use `RLP_AUTH_TAG_SIZE -1`?  Or do something like this:

```java
  public static final int TAG_SIZE = 32;
  public static final int RLP_AUTH_TAG_PREFIX_SIZE = 1;
  public static final int AUTH_TAG_SIZE = 12;
  public static final int RLP_AUTH_TAG_SIZE = AUTH_TAG_SIZE + RLP_AUTH_TAG_PREFIX_SIZE;
  public static final int HEADER_SIZE = TAG_SIZE + RLP_AUTH_TAG_SIZE;

  private final Bytes tag;
  private final Bytes authTag;
  private final Bytes payload;

  public static TaggedMessage decode(Bytes bytes) {
    Bytes tag = Bytes.wrap(bytes.slice(0, TAG_SIZE));
    Bytes authTagRlpBytes = bytes.slice(TAG_SIZE, RLP_AUTH_TAG_SIZE);
    Bytes authTag = decodeSingleRlpString(authTagRlpBytes);
    if (authTag.size() != AUTH_TAG_SIZE) {
```

Reply:
Oh, right, thanks 👍